lib/assets/javascripts/twine.js.coffee

```diff
@@ -194,7 +203,7 @@ wrapFunctionString = (code, args, node) ->
       ($context, $root) -> getValue($context, keypath)
   else
     try
-      new Function(args, "with($context) { return #{code} }")
+      new Function(args, "with($context) { with($modules) { return #{code} } }")
```

Comment:
Rather than changing the function string above to do some conditionals it was easier to pass in an empty object literal for `$modules` for all but `define` & `eval`. Seemed like the nicer fix, any other suggestions are more than welcome 👍